net/quagga/src/opnsense/mvc/app/controllers/OPNsense/Quagga/Api/Ospf6settingsController.php

```diff
@@ -0,0 +1,195 @@
+                // return node
+                return array("interface" => $node->getNodes());
+            }
+        } else {
+            $node = $mdlOSPF6->interfaces->interface->add();
+            return array("interface" => $node->getNodes());
+        }
+        return array();
+    }
+    public function addInterfaceAction()
+    {
+        $result = array("result" => "failed");
+        if ($this->request->isPost() && $this->request->hasPost("interface")) {
+            $result = array("result" => "failed", "validations" => array());
+            $mdlOSPF6 = $this->getModel();
+            $node = $mdlOSPF6->interfaces->interface->Add();
+            $node->setNodes($this->request->getPost("interface"));
+            $valMsgs = $mdlOSPF6->performValidation();
+            foreach ($valMsgs as $field => $msg) {
+                $fieldnm = str_replace($node->__reference, "interface", $msg->getField());
+                $result["validations"][$fieldnm] = $msg->getMessage();
+            }
+            if (count($result['validations']) == 0) {
+                // save config if validated correctly
+                $mdlOSPF6->serializeToConfig();
+                Config::getInstance()->save();
+                unset($result['validations']);
+                $result["result"] = "saved";
+            }
+        }
+        return $result;
+    }
+    public function delInterfaceAction($uuid)
+    {
+        $result = array("result" => "failed");
+        if ($this->request->isPost()) {
+            $mdlOSPF6 = $this->getModel();
+            if ($uuid != null) {
+                if ($mdlOSPF6->interfaces->interface->del($uuid)) {
+                    $mdlOSPF6->serializeToConfig();
+                    Config::getInstance()->save();
+                    $result['result'] = 'deleted';
+                } else {
+                    $result['result'] = 'not found';
+                }
+            }
+        }
+        return $result;
+    }
+    public function setInterfaceAction($uuid)
+    {
+        if ($this->request->isPost() && $this->request->hasPost("interface")) {
+            $mdlNetwork = $this->getModel();
+            if ($uuid != null) {
+                $node = $mdlNetwork->getNodeByReference('interfaces.interface.' . $uuid);
+                if ($node != null) {
+                    $result = array("result" => "failed", "validations" => array());
+                    $interfaceInfo = $this->request->getPost("interface");
+                    $node->setNodes($interfaceInfo);
+                    $valMsgs = $mdlNetwork->performValidation();
+                    foreach ($valMsgs as $field => $msg) {
+                        $fieldnm = str_replace($node->__reference, "interface", $msg->getField());
+                        $result["validations"][$fieldnm] = $msg->getMessage();
+                    }
+                    if (count($result['validations']) == 0) {
+                        // save config if validated correctly
```

Comment:
I am not sure if you need to unset() the 'validations' here. this is why it originally did not close the modal.